presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixTableProperties.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.phoenix;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.spi.connector.ColumnMetadata;
+import io.prestosql.spi.session.PropertyMetadata;
+import io.prestosql.spi.type.TypeManager;
+import org.apache.hadoop.util.StringUtils;
+
+import javax.inject.Inject;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.prestosql.spi.session.PropertyMetadata.booleanProperty;
+import static io.prestosql.spi.session.PropertyMetadata.integerProperty;
+import static io.prestosql.spi.session.PropertyMetadata.stringProperty;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Class contains all table properties for the Phoenix connector. Used when creating a table:
+ * <p>
+ * CREATE TABLE foo (a VARCHAR with (primary_key = true), b INT) WITH (SALT_BUCKETS=10, VERSIONS=5, COMPRESSION='lz');
+ */
+public final class PhoenixTableProperties
+{
+    private final List<PropertyMetadata<?>> tableProperties;
+    public static final String ROWKEYS = "rowkeys";
+    public static final String SALT_BUCKETS = "salt_buckets";
+    public static final String SPLIT_ON = "split_on";
+    public static final String DISABLE_WAL = "disable_wal";
+    public static final String IMMUTABLE_ROWS = "immutable_rows";
+    public static final String DEFAULT_COLUMN_FAMILY = "default_column_family";
+    public static final String BLOOMFILTER = "bloomfilter";
+    public static final String VERSIONS = "versions";
+    public static final String MIN_VERSIONS = "min_versions";
+    public static final String COMPRESSION = "compression";
+    public static final String TTL = "ttl";
+
+    private final List<PropertyMetadata<?>> columnProperties;
```

Comment:
Let's split this out into a `PhoenixColumnProperties` class to avoid confusion about why column properties are in the table properties class